Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h

```diff
@@ -35,21 +35,24 @@ class CGDisplayList {
     WTF_MAKE_NONCOPYABLE(CGDisplayList);
 public:
     CGDisplayList() = default;
-    explicit CGDisplayList(WebCore::SharedBuffer& displayList)
+    CGDisplayList(WebCore::SharedBuffer& displayList, Vector<MachSendRight>&& surfaces)
         : m_displayList(&displayList)
+        , m_surfaces(surfaces)
     {
     }
 
     CGDisplayList(CGDisplayList&&) = default;
     CGDisplayList& operator=(CGDisplayList&&) = default;
 
     RefPtr<WebCore::SharedBuffer> buffer() const { return m_displayList; }
+    Vector<MachSendRight>&& takeSurfaces() { return WTFMove(m_surfaces); }
```

Comment:
WTFMove or `std::exchange` I feel like the latter is better.

Reply:
True, good to leave something sane behind.

Reply:
This function is not good, it doesn't take anything. It allowed the caller to take.

Should either be:
```
Vector<MachSendRight> takeSurfaces() { return WTFMove(m_surfaces); }
```
or
```
Vector<MachSendRight>&& takeSurfaces() { return std::exchange(m_surfaces, Vector<MachSendRight>()); }
```

I recommend the former.